components/files/MainFiles.tsx

```diff
@@ -411,6 +413,20 @@ export default function MainFiles(
     moveDirectoryOrFileModal.value = null;
   }
 
+  function onClickDownloadDirectory(parentPath: string, name: string) {
+    // Create download URL with proper path encoding
+    const fullPath = parentPath + name + '/';
```

Comment:
Please send `parentPath` and `name` to the endpoint instead of `path`, like for other endpoints, so the proper joining/resolving/parsing is done in the backend.